### ISSUE TITLE
[Refactor] Shift the audio playing logic for the Effects to the useAudioPlayer custom hook

### ISSUE THREAD

**Issue by ArvindParekh** [ArvindParekh/Lofi-Music-App#15]
Currently, the audio-playing logic for the Tracks is consolidated within the [useAudioPlayer](https://github.com/ArvindParekh/Lofi-Music-App/blob/main/src/lib/useAudioPlayer.js) custom hook. However, the same logic for Effects is not centralized and is present individually within the `Effect.jsx` file. To maintain a consistent and centralized source of truth, it's advisable to include the audio playing logic for Effects within the useAudioPlayer custom hook.

The task involves:

- Modifying the useAudioPlayer custom hook to incorporate the audio playing logic for Effects.
- Adjusting the `Effects.jsx` and `EffectCard.jsx` components to accommodate this change, ensuring there's no loss of functionality.

**Comment by tigerabrodi** [comment]
@ArvindParekh 

Do you mean `EffectCard.jsx` should also use the `useAudioPlayer` hook just like Tracks file does?

**Comment by ArvindParekh** [comment]
Yes, precisely. However, the `useAudioPlayer` hook currently doesn't support the kind of logic that's required for playing "Effect" audios, the logic is still present natively in `EffectCard.jsx`. It'd be better if there's one single custom hook that takes care of the entire audio logic for both Effects and Tracks.

**Comment by sahild91** [comment]
@ArvindParekh ,
Is this issue still open for contribution?
If yes, maybe I can try it out and see if it can be made as a custom hook.

**Comment by ArvindParekh** [comment]
Hey @sahild91, yes the issue's still open, feel free to work on it if you'd like. We already have a custom hook in place, we'd just like to incorporate the logic of Effects into it.
(You can get the issue assigned to you, and here are the [contributing guidelines](https://github.com/ArvindParekh/Lofi-Music-App/blob/main/CONTRIBUTING.md) if you'd like to take a look 👍🏻 )

**Comment by qu1ll** [comment]
Hey @sahild91, just wondering how far you got on this issue, I was thinking of taking a peak at it see what I can do.
